### PR TITLE
ft(Chat) user should be able to chat[#187364820]

### PR DESCRIPTION
**Pull request by tuyishimehope**
## ℹ️ Description

Please write about the changes implemented in this PR. Also write a short description about why the change is implemented.

##### 📹 Loom video link:
https://www.loom.com/share/dabe51649db144bb8244e8b76a9d4f5c

## 🧪 How can this be tested?

Write a short description on how others can test the changes implemented in the PR

## 📸 Screenshots

image
image
image

## 📍 Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shebz2023 — COMMENTED** [review]
Good work @tuyishimekyrie  ,write some tests and  make the hover color on the sidebar look like the one of the selected

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

File: src/pages/rootpages/Cart.tsx [hunk 1]
```diff
@@ -64,6 +64,28 @@ const Cart: React.FC = () => {
...
+  //                     "https://example.com",
+  //                     "https://example.com",
+  //                     "https://example.com",
+  //                     "https://example.com"
+  //                 ]
+  //             }
+  //         ],
+  //         "totalPrice": 4800
+  //     }
+  // }
```
Comment (SergeInGithub) [inline src/pages/rootpages/Cart.tsx]:
```suggestion
```

File: src/pages/userpage/Chat.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    }
+    const hours = date.getHours() % 12 || 12;
+    const minutes = date.getMinutes().toString().padStart(2, '0');
+    const amPm = date.getHours() < 12 ? 'AM' : 'PM';
+    return `${hours}:${minutes} ${amPm}`;
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen text-white w-full ">
+      <div className="w-full  rounded-lg   ">
```
Comment (SergeInGithub) [inline src/pages/userpage/Chat.tsx]:
```suggestion
      <div className="w-full  rounded-lg">
```

File: src/pages/userpage/Chat.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            {messages.map((msg, index) => (
+              <li
+                key={index}
+                className={`mb-8 p-2 rounded-lg  ${
+                  msg.senderId === localStorage.getItem('userId')
+                    ? ' text-green-500 text-right text-xl'
+                    : ' text-black text-left  text-xl'
+                }`}
+              >
+                <strong className="">
```
Comment (SergeInGithub) [inline src/pages/userpage/Chat.tsx]:
```suggestion
                <strong>
```

File: src/pages/userpage/Chat.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    const minutes = date.getMinutes().toString().padStart(2, '0');
+    const amPm = date.getHours() < 12 ? 'AM' : 'PM';
+    return `${hours}:${minutes} ${amPm}`;
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen text-white w-full ">
+      <div className="w-full  rounded-lg   ">
+        <div className="main-container " id="chatApp">
+          <ul id="messages" className=" max-w-full  overflow-y-auto  h-[90vh] ">
```
Comment (SergeInGithub) [inline src/pages/userpage/Chat.tsx]:
```suggestion
          <ul id="messages" className=" max-w-full  overflow-y-auto  h-[90vh]">
```

File: src/pages/userpage/Chat.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      return 'Invalid Date';
+    }
+    const hours = date.getHours() % 12 || 12;
+    const minutes = date.getMinutes().toString().padStart(2, '0');
+    const amPm = date.getHours() < 12 ? 'AM' : 'PM';
+    return `${hours}:${minutes} ${amPm}`;
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen text-white w-full ">
```
Comment (SergeInGithub) [inline src/pages/userpage/Chat.tsx]:
```suggestion
    <div className="flex flex-col items-center justify-center min-h-screen text-white w-full">
```

File: src/pages/userpage/Chat.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+            onSubmit={handleSend}
+            className="flex absolute bottom-0 mb-4 ml-2 w-[65vw]"
+          >
+            <input
+              id="input"
+              placeholder="Enter a message"
+              value={message}
+              onChange={handleInputChange}
+              onKeyDown={handleKeyPress}
+              className="flex-grow p-2 border border-gray-400 rounded-l-lg  text-black"
```
Comment (SergeInGithub) [inline src/pages/userpage/Chat.tsx]:
```suggestion
              className="flex-grow p-2 border border-gray-400 rounded-l-lg text-black"
```

File: src/pages/userpage/Home.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        </div>
+        <div className="h-56 w-72">
+          <img
+            src={images}
+            alt="vision"
+            className="w-full h-full object-cover"
+          />
+        </div>
+      </div>
+      {/* <h1 className="px-4">New Trend Style</h1> */}
```
Comment (SergeInGithub) [inline src/pages/userpage/Home.tsx]:
```suggestion
```

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Nice work so far @tuyishimekyrie You can try and improve the design of the chat a little. For example you can wrap each message in a card instead of just text. make the cards of the receiver and sender a different color and you can improve  Also, work on the suggestions below

**Review by SergeInGithub — DISMISSED** [review]
LGTM. Just rebase and you're good to go

File: src/pages/userPages/Chat.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+              onChange={handleInputChange}
+              onKeyDown={handleKeyPress}
+              className="flex-grow p-2 border border-gray-400 rounded-l-lg text-black"
+            />
+            <button
+              type="submit"
+              className="p-2 bg-green-400 text-black rounded-r-lg"
+            >
+              Send
+            </button>
```
Comment (SergeInGithub) [inline src/pages/userPages/Chat.tsx]: we have a button component

File: src/pages/userPages/Home.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+          <h4 className="text-slate-200 text-xl py-4">
+            change your daily life with a modern lifestyle
+          </h4>
+          <button
+            className="px-4 py-2 hover:text-white flex items-center justify-between gap-4 text-black bg-slate-200  rounded-md hover:bg-transparent"
+            onClick={() => navigate('/shop')}
+          >
+            Check Now
+            <CiLocationArrow1 />
+          </button>
```
Comment (SergeInGithub) [inline src/pages/userPages/Home.tsx]: we have an existing button component

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Work on the reviews provided below @tuyishimekyrie

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**